src/Tuura/Plato/Translation.hs

```diff
@@ -82,23 +82,20 @@ validateInterface signs circuit
     unused       = filter ((==Unused) . interface circuit) signs
 
 cartesianProduct :: Ord a => NonEmpty.NonEmpty [a] -> [[a]]
-cartesianProduct l = removeSupersets sortByLength
+cartesianProduct l = removeSupersets sortAllLists
   where
     sequenced    = sequence (NonEmpty.toList l)
     removeDupes  = map nub sequenced
     removeNull   = filter (not . null) removeDupes
     sortAllLists = map sort removeNull
-    sortByLength = sortBy (comparing length) sortAllLists
 
 removeSupersets :: Eq a => [[a]] -> [[a]]
 removeSupersets s = filter (not . null) result
   where
     prev n        = take n s
```

Comment:
Oh no I did remove these and then I reverted. 

I'm also working on the flipping thing we talked about, mixing branches :S